nltk/draw/tsne.py

```diff
@@ -0,0 +1,173 @@
+#
+#  tsne.py
+#  
+# Implementation of t-SNE in Python. The implementation was tested on Python 2.5.1, and it requires a working 
+# installation of NumPy. In order to plot the results of this example, a working installation of matplotlib is required.
+# The example can be run by executing: ipython tsne.py -pylab
+#
+#
+#  Created by Laurens van der Maaten on 20-12-08.
+#  Copyright (c) 2008 Tilburg University. All rights reserved.
```

Comment:
+1 for scikit-learn. It's a wide-spread and actively developed library, so using it will make NLTK maintenance much simpler.